Source/Core/Core/Core.cpp

```diff
@@ -819,6 +828,65 @@ void RunAsCPUThread(std::function<void()> function)
     PauseAndLock(false, was_unpaused);
 }
 
+void RunOnCPUThread(std::function<void()> function, bool wait_for_completion)
+{
+  // If the CPU thread is not running, assume there is no active CPU thread we can race against.
+  // Therefore, we can safely exeucte the callback on the calling thread. TODO: Is this true?
+  if (!IsRunning() || CPU::IsStepping() || IsCPUThread())
```

Comment:
Separate out IsCPUThread() into a separate if statement.

Taking a savestate while paused or stepping is a common operation for making TASes, and we definitely want to be saving EFB state for those.

Perhaps there's a way you could wake up a sleeping CPU thread when a task is enqueued?

Reply:
Hmm, good point. Didn't think of that use case. I'll think about how I can work around it.